firmware/repository.py

```diff
@@ -0,0 +1,126 @@
+"""Repository - module to communicate with the database"""
+
+from firmware.database import db_session
+from firmware.models import User, Company, Category, Review
+from sqlalchemy import and_
+
+
+def get_companies():
+    """ gets all companies from database """
+    companies = db_session.query(Company)
+    return companies
+
+
+def get_user_id(username):
+    """ gets a user id via it's unique username """
+    user_id = db_session.query(User.id).filter(User.username == username)
+    return user_id
+
+
+def add_company(company):
+    """ adds a company to the database """
+    db_session.add(company)
+    db_session.commit()
+    return company.id
+
+
+def get_company(company_id):
+    """ retrieves a specific company, via it's id """
+    company = db_session.query(Company).filter(Company.id == company_id).one()
+    if company is None:
+        return None
+    return company
+
+
+def get_category(company_id):
+    """ gets a category for a specific company """
+    company = db_session.query(Company).filter(Company.id == company_id).one()
+    category = db_session.query(Category).filter(company.category_id ==
+                                                 Category.id).one()
+    return category.domain
+
+
+def get_categories():
+    """ gets all categories from the database """
+    categories = db_session.query(Category).all()
+    return categories
+
+
+def get_reviews(company_id):
+    """ gets all reviews for a company """
+    reviews = db_session.query(Review).filter(Review.company_id == company_id)\
+                .order_by(Review.id.desc()).all()
+    return reviews
+
+
+def add_user(user):
+    """ adds a new user to the database """
+    db_session.add(user)
+    db_session.commit()
+    return user.id, user.username
+
+
+def get_user(username):
+    """gets a user by unique username """
+    user = db_session.query(User).filter(User.username == username).one().serialize()
+    return user
+
+
+def check_user(username, password):
+    """ checks if the user exists in the database, with the entered password """
+    compatible_user = db_session.query(User).filter(and_(User.username == username,
+                                                         User.password == password)).all()
+    if not compatible_user:
+        return 0
+    return 1
+
+
+def add_reviews(review):
```

Comment:
This method adds only one review, therefore you should use singular in the method name.